lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -337,6 +337,13 @@ LogicalResult CircuitOp::verifyRegions() {
     return failure();
   }
 
+  // Even though ClassOps are FModuleLike, they are not a hardware entity, so
+  // we ban them from being our top-module in the design.
+  if (isa<ClassOp>(mainModule)) {
+    emitOpError("must have a non-class top module");
```

Comment:
nit: InFlightDiagnostic should implicitly convert to failure, so I think you can `return emitOpError(...)`